docker/Dockerfile

```diff
@@ -161,68 +162,48 @@ RUN --mount=type=bind,source=tower-batch-control,target=tower-batch-control \
     --mount=type=cache,target=${APP_HOME}/target/ \
     --mount=type=cache,target=/usr/local/cargo/git/db \
     --mount=type=cache,target=/usr/local/cargo/registry/ \
-cargo build --locked --release --features "${FEATURES}" --package zebrad --bin zebrad && \
-cp ${APP_HOME}/target/release/zebrad /usr/local/bin
+    cargo build --locked --release --features "${FEATURES}" --package zebrad --bin zebrad && \
+    cp ${APP_HOME}/target/release/zebrad /usr/local/bin
 
-# This stage is only used when deploying nodes or when only the resulting zebrad binary is needed
-#
-# To save space, this step starts from scratch using debian, and only adds the resulting
-# binary from the `release` stage
+# This step starts from scratch using Debian and only adds the resulting binary
+# from the `release` stage.
 FROM debian:bookworm-slim AS runtime
 
-# Set the default path for the zebrad binary
-ARG APP_HOME
-ENV APP_HOME=${APP_HOME}
-WORKDIR ${APP_HOME}
```

Comment:
We should set a `WORKDIR` otherwise the user will end up in the `/` directory with no permissions, which they might require for testing purposes.

The `ARG + ENV` combination also allows the user to set a custom directory, in case their host permissions does not allow the one we've chosen.

Reply:
<img width="380" alt="image" src="https://github.com/user-attachments/assets/4ff7c2b7-5f03-48df-b4d7-f47d48410b3e">

Reply:
System users should have no home dir and should not even be able to log in to the machine. Our users should use `-u root` when they logging to the machine. The logic I had in mind is very simple and minimal:

- The whole `runtime` target has its entrypoint executed under the non-privileged `zebra` system user which has no home dir and no login (just as in a regular Linux env).
- The Dockerfile sets up the minimal requirements for the `zebra` user to execute the entrypoint.
- When our user wants to do some tweaks, they explicitly use `-u root`, which they can always do, and which gives them clarity that they have full privileges.

I wanted to describe this in our docs in a subsequent PR. Is it OK if we do it like that?